src/openms/source/ANALYSIS/QUANTITATION/AbsoluteQuantitation.cpp

```diff
@@ -641,5 +641,58 @@ namespace OpenMS
     }
   } 
 
-} // namespace
+  void AbsoluteQuantitation::getOptimizedCalibrationCurve(
+    const String& component_name,
+    std::vector<AbsoluteQuantitationStandards::featureConcentration>& component_concentrations
+  )
+  {
+    std::sort(component_concentrations.begin(), component_concentrations.end(),
+      [](AbsoluteQuantitationStandards::featureConcentration lhs, AbsoluteQuantitationStandards::featureConcentration rhs)
+      {
+        return lhs.actual_concentration < rhs.actual_concentration; //ascending order
+      }
+    );
+    std::map<String, AbsoluteQuantitationMethod>::iterator qm = quant_methods_.find(component_name);
```

Comment:
This is a good idea to improve the efficiency of the method by saving the loop over all component_names in the quant_methods_ map.  However, if we use my suggestions above, this will not be needed.

Reply:
Also, if you change the `LOG_INFO << "Warning: Standards not found for component " << quant_method.first << ".";` on line 639 to `LOG_DEBUG << "Warning: Standards not found for component " << quant_method.first << ".";` to suppress the output to the console that would be greatly appreciated.